shell/platform/common/cpp/json_method_codec.cc

```diff
@@ -9,9 +9,27 @@
 namespace flutter {
 
 namespace {
+
 // Keys used in MethodCall encoding.
 constexpr char kMessageMethodKey[] = "method";
 constexpr char kMessageArgumentsKey[] = "args";
+
+// Returns a new document containing only |element|, which must be an element
+// in |document|. This is a move rather than a copy, so it is efficient but
+// destructive to the data in |document|.
+std::unique_ptr<rapidjson::Document> ExtractElement(
+    rapidjson::Document* document,
+    rapidjson::Value* subtree) {
+  auto extracted = std::make_unique<rapidjson::Document>();
+  // Pull the subtree up to the root of the document.
+  document->Swap(*subtree);
+  // Swap the entire document into |extracted|. Unlike the swap above this moves
+  // the allocator ownership, so the data won't be deleted when |document| is
+  // destroyed goes out of scope.
```

Comment:
"destroyed when it goes out of scope" ?

Reply:
Just "destroyed". "goes out of scope" was from the older comment, where the object was a local variable, rather than a pointer (so going out of scope doesn't have any effect), but I screwed up the rewrite.